exporter/splunkhecexporter/worker_queue.go

```diff
@@ -0,0 +1,121 @@
+package splunkhecexporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+var errSendingQueueIsFull = errors.New("worker_queue is full")
+
+type request struct {
+	bytes              []byte
+	localHeaders       map[string]string
+	compressionEnabled bool
+}
+
+type WorkerQueue interface {
+	Start() error
+	Stop() error
+	Produce(*bufferState, map[string]string) error
+}
+
+type workerQueue struct {
+	config  *Config
+	workers []hecWorker
+	queue   chan *request
+	wg      sync.WaitGroup
+	logger  *zap.Logger
+}
+
+func newWorkerQueue(config *Config, logger *zap.Logger, clientFunc func() *http.Client) *workerQueue {
+
+	workers := make([]hecWorker, 0, config.WorkerQueueConfig.Worker)
+	for i := 0; i < config.WorkerQueueConfig.Worker; i++ {
+		workers = append(workers, newHecWorkerWithoutAck(config, clientFunc))
+	}
+	queue := make(chan *request, config.WorkerQueueConfig.Size)
+
+	return &workerQueue{
+		config:  config,
+		workers: workers,
+		queue:   queue,
+	}
+}
+
+func (l *workerQueue) Start() error {
+	if l == nil {
+		return fmt.Errorf("uninitialised worker queue")
+	}
+	l.wg.Add(l.config.WorkerQueueConfig.Worker)
+	for _, worker := range l.workers {
+		go l.consume(worker)
+	}
+	return nil
+}
+
+func (l *workerQueue) consume(worker hecWorker) {
+	defer l.wg.Done()
+CONSUMER_LOOP:
+	for request := range l.queue {
+		if !l.config.RetrySettings.Enabled {
+			if err := worker.Send(context.Background(), request); err != nil {
+				l.logger.Error("Failed to send data to splunk", zap.Error(err))
+				continue
+			}
+		}
+
+		var err error
+		for i := 0; i < l.config.WorkerQueueConfig.Retries; i++ {
```

Comment:
There should not be a retry mechanism in the exporter, I think retries are done at the pipeline level by the collector.